lib/puppet/parser/functions/docker_run_flags.rb

```diff
@@ -5,17 +5,27 @@
 # docker_run_flags.rb
 #
 module Puppet::Parser::Functions
+  newfunction(:'docker::escape', type: :rvalue) do |args|
+    subject = args[0]
+
+    if self['facts'] && self['facts']['os']['family'] == 'windows'
+      %("#{subject.gsub('"', '\\"')}")
```

Comment:
The escaping on line 68 suggest PowerShell escaping is used (backtick at end of line while cmd.exe would use a `^`):

https://github.com/puppetlabs/puppetlabs-docker/blob/355b5b33714cd82f91e24bf400aefde2515b1a74/lib/puppet/parser/functions/docker_run_flags.rb#L68

so quotes escaping above should be wrong… but I can't give it a try on windows ATM